### PR TITLE
New documentation using soss. [8217]

### PR DESCRIPTION
**Pull request by LuisGP**
Documentation using soss.

### CODE REVIEW

File: docs/index.rst [hunk 1]
```diff
@@ -10,22 +10,49 @@ eProsima Integration Service Documentation
...
-
-|
-| The main features of *Integration Service* are:
-
-* Connects two different domains.
-* Mapping between different data types.
-* User-defined operations over the received messages (:ref:`transformation library`).
-* Communication with other environments, like *ROS2*.
+*eProsima Integration-Services* is a tool based on `SOSS <https://github.com/eProsima/soss_v2>`__ and its
+**System-Handle** `SOSS-DDS <https://github.com/eProsima/SOSS-DDS>`__ to allow intercommunicating any
```
Comment (jamoralp) [inline docs/index.rst:14]: `to allow intercommunication between a DDS-based system and any other protocol-based system, including other DDS systems; as well as offering the possibility to integrate them into a larger, even more complex system.`

File: docs/index.rst [hunk 2]
```diff
@@ -10,22 +10,49 @@ eProsima Integration Service Documentation
...
+*DDS* based system with any other protocol, including other *DDS* systems, integrate them into a larger system,
+even more complex system.
+
+Here is the minimal example, which translates a single topic from ROS 2 to DDS:
```
Comment (jamoralp) [inline docs/index.rst:18]:
```suggestion
Here is a minimal example, which translates a single topic from ROS 2 to DDS:
```

File: docs/index.rst [hunk 3]
```diff
@@ -10,22 +10,49 @@ eProsima Integration Service Documentation
...
+
+.. code-block:: yaml
+
+    systems:
+        ros2: { type: ros2 }
+        dds: { types-from: ros2 }
+    topics:
+        chatter: { type: std_msgs/String, route: {from: ros2, to: dds} }
+
+The intent is that different translations are possible by only changing the configuration file. For example, by changing
```
Comment (jamoralp) [inline docs/index.rst:28]:
```suggestion
The aim is that different translations are possible by only changing the configuration file. For example, by changing
```

File: docs/getting_started.rst [hunk 4]
```diff
@@ -3,150 +3,54 @@ Getting Started
...
-
-.. code-block:: xml
+Installation
+^^^^^^^^^^^^
 
-    <connector name="domain_change">
-        <reader participant_profile="domain0" subscriber_profile="is_subscriber"/>
-        <writer participant_profile="domain5" publisher_profile="is_publisher"/>
-    </connector>
+To build `integration-services` it is recommended use a
```
Comment (jamoralp) [inline docs/getting_started.rst:13]:
```suggestion
To build `integration-services` it is recommended to use a
```

File: docs/getting_started.rst [hunk 5]
```diff
@@ -3,150 +3,54 @@ Getting Started
...
+`colcon workspace <https://colcon.readthedocs.io/en/released/user/quick-start.html>`__.
+The `integration-services` repo consists of many cmake packages which can be configured and built manually,
+but colcon makes the job much smoother.
 
-To execute the example properly, we must first compile the example itself, from the
-`domain_change example location <https://github.com/eProsima/Integration-Service/tree/master/examples/domain_change>`__.
+We will assume that you have installed
+`Fast-RTPS <https://github.com/eProsima/Fast-RTPS/>`__ using any system-wide method, and
+`soss <https://github.com/eProsima/soss_v2>`__ following its installation manual, and source it,
```
Comment (jamoralp) [inline docs/getting_started.rst:20]:
```suggestion
`soss <https://github.com/eProsima/soss_v2>`__ following its installation manual; and sourced it,
```

File: docs/getting_started.rst [hunk 6]
```diff
@@ -3,150 +3,54 @@ Getting Started
...
-both DomainChange instances will begin to communicate.
-
-In another terminal:
-
-.. code-block:: bash
+    source install/setup.bash
 
-    $ cd <path_to_is_source>/examples/domain_change
-    $ integration_service config.xml
+From now, `soss` should be able to locate `integration-service` (`SOSS-DDS`) **System-Handle**.
```
Comment (jamoralp) [inline docs/getting_started.rst:53]:
```suggestion
From now on, `soss` should be able to locate `integration-service` (`SOSS-DDS`) **System-Handle**.
```

File: docs/use_cases.rst [hunk 7]
```diff
@@ -0,0 +1,83 @@
+Use cases
+=========
+
+In this section the most representative use-cases are shown:
+
+
+DDS Bridge
+^^^^^^^^^^
+
+A typical scenario when communicating different `DDS` systems is to face that these systems use incompatible
```
Comment (jamoralp) [inline docs/use_cases.rst:10]:
```suggestion
A typical scenario when communicating different `DDS` systems is to find out that these systems use incompatible
```

File: docs/use_cases.rst [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+configuration, for example, `DDS`'s HelloWorld and `ROS 2`'s Talker-Listener.
```
Comment (jamoralp) [inline docs/use_cases.rst:11]:
```suggestion
configuration; for example, `DDS`'s HelloWorld and `ROS 2`'s Talker-Listener.
```

File: docs/use_cases.rst [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+
+.. image:: DDS_NO_COMS.png
+
+A user with knowledge of both systems may know that `ROS 2` uses `DDS` as middleware but hides some
+configuration details making difficult, or even impossible a direct communication.
```
Comment (jamoralp) [inline docs/use_cases.rst:16]:
```suggestion
configuration details, making it difficult, or even impossible, to establish a direct communication.
```

File: docs/use_cases.rst [hunk 10]
```diff
@@ -0,0 +1,83 @@
...
+By using `integration-service` this communication can be achieved with minimal effort from the user.
+
+.. image:: DDS_WITH_IS.png
+
+See :ref:`ROS2` for an example of this use-case.
+
+Integrate a large system
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+Most systems evolve with the time, adding new functionality or new components.
```
Comment (jamoralp) [inline docs/use_cases.rst:27]:
```suggestion
Most systems evolve over time, adding new functionality or new components.
```

File: docs/use_cases.rst [hunk 11]
```diff
@@ -0,0 +1,83 @@
...
+When these new components are software or hardware, that doesn't use a protocol compatible with the rest
+of the system, we are forced to create an additional component usually known as *bridge*.
+
+If the system contains several subsystems, and each component use a different protocol, a *bridge* must be
+created for each existing component pair that must communicate.
```
Comment (jamoralp) [inline docs/use_cases.rst:32]:
```suggestion
If the system contains several subsystems, and each component uses a different protocol, a *bridge* must be
created for each existing component pair that must have communication capabilities.
```

File: docs/examples.rst [hunk 12]
```diff
@@ -0,0 +1,210 @@
+Examples
+========
+
+ROS2
+^^^^
+
+This examples ties to illustrate the :ref:`DDS Bridge` use-case.
```
Comment (jamoralp) [inline docs/examples.rst:7]:
```suggestion
This examples tries to illustrate the :ref:`DDS Bridge` use-case.
```

File: docs/examples.rst [hunk 13]
```diff
@@ -0,0 +1,210 @@
...
+
+    ros2 run demo_nodes_cpp talker
+
+- In the second terminal, execute the `Fast-RTPS` HelloWorld subscriber
+
+.. code-block:: bash
+
+    ./HelloWorldExample subscriber
+
+Both application don't communicate due to the incompatibility of their `DDS` configuration, **topic** and **type**.
```
Comment (jamoralp) [inline docs/examples.rst:32]:
```suggestion
Both applications don't communicate due to the incompatibility of their `DDS` configuration, **topic** and **type**.
```

File: docs/examples.rst [hunk 14]
```diff
@@ -0,0 +1,210 @@
...
+
+- In the third terminal, execute `integration-service` with the `someip_dds.yaml` configuration file:
+
+.. code-block:: bash
+
+    soss examples/radar_fusion_dds/someip_dds.yaml
+
+`radar` and `RadarExample` subscriber should start communicating.
+
+If you want to test the other way, launc `fusion`, `RadarExample` as publisher, and `integration-service` with the file
```
Comment (jamoralp) [inline docs/examples.rst:143]:
```suggestion
If you want to test the other way, launch `fusion`, `RadarExample` as publisher, and `integration-service` with the file
```

**Comment by LuisGP** [comment]
Solved all conversations because they become obsolete.

**Pull request merged**